modules/veteran/app/models/veteran/service/representative.rb

```diff
@@ -1,12 +1,53 @@
 # frozen_string_literal: true
 
+require 'csv'
+
 module Veteran
   # Not technically a Service Object, this is a term used by the VA internally.
   module Service
     class Representative < ActiveRecord::Base
+      BASE_URL = 'https://www.va.gov/ogc/apps/accreditation/'
+
       self.primary_key = :representative_id
 
       validates_presence_of :poa
+
+      def self.reload!
+        array_of_hashes = fetch_data('orgsexcellist.asp')
+        array_of_hashes.each do |hash|
+          representative = Representative.find_or_initialize_by(representative_id: hash['Registration Num'])
+          representative.poa = hash['POA'].gsub!(/\W/, '')
+          representative.first_name = hash['Representative'].split(' ').second
+          representative.last_name = hash['Representative'].split(',').first
+          representative.phone = hash['Org Phone']
+          representative.save
+        end
+
+        Representative.where.not(representative_id: array_of_hashes.map { |h| h['Registration Num'] }).destroy_all
+
+        array_of_organizations = array_of_hashes.map do |h|
+          { poa: h['POA'], name: h['Organization Name'], phone: h['Org Phone'], state: h['Org State'] }
+        end.uniq.compact
+
+        array_of_organizations.each do |hash|
+          Organization.find_or_create_by(hash)
+        end
+      end
+
+      def self.fetch_data(action)
+        page = Faraday.new(url: BASE_URL).post(action, id: 'frmExcelList', name: 'frmExcelList').body
+        doc = Nokogiri::HTML(page)
+        content = CSV.generate(headers: true) do |csv|
+          doc.xpath('//table/tr').each do |row|
+            tarray = []
+            row.xpath('td').each do |cell|
+              tarray << cell.text
+            end
+            csv << tarray
+          end
+        end
+        CSV.parse(content, headers: :first_row).map(&:to_h)
```

Comment:
so you're building a CSV then parsing it and calling `to_h` feels like you could just map over the `doc.xpath` creating hashes along the way... removing the csv portion all together.

Reply:
I thought about that but he problem that I ran into is that I wanted to account for changes they make on their end and since their ASP page is designed to work like a csv with the heads at the top, the cleanest implementation I could think of was to go from HTML to CSV and then to hash where the headers are saved accordingly.